wfl/fit/ace.py

```diff
@@ -0,0 +1,299 @@
+                    # elif suffix == '.yace' or suffix == '.yaml':
+                        # _ = yaml.safe_load(fin)
+                    elif suffix == '.size':
+                        # dry run results
+                        size = _read_size()
+                    else:
+                        warnings.warn('Cannot parse ACE file with suffix '+suffix)
+        except:
+            need_to_fit = True
+
+        if dry_run and not need_to_fit:
+            return size
+
+        if not need_to_fit:
+            return ace_file_base
+
+    if need_to_fit and remote_info is not None and remote_info != '_IGNORE':
+        input_files = remote_info.input_files.copy()
+        output_files = remote_info.output_files.copy()
+
+        # put configs in memory so they can be staged out easily
+        fitting_configs = fitting_configs.in_memory()
+
+        # run dir will contain only things created by fitting, so it's safe to copy the
+        # entire thing back as output
+        output_files.append(str(run_dir))
+
+        xpr = ExPyRe(name=remote_info.job_name, pre_run_commands=remote_info.pre_cmds, post_run_commands=remote_info.post_cmds,
+                      env_vars=remote_info.env_vars, input_files=input_files, output_files=output_files, function=fit,
+                      kwargs= {'fitting_configs': fitting_configs, 'ACE_name' : ACE_name, 'params': params,
+                               'ref_property_prefix': ref_property_prefix,
+                               'run_dir': str(run_dir), 'formats': formats, 'ace_fit_exec': ace_fit_exec,
+                               'dry_run': dry_run, 'verbose': verbose, 'remote_info': '_IGNORE'})
+
+        xpr.start(resources=remote_info.resources, system_name=remote_info.sys_name,
+                  exact_fit=remote_info.exact_fit, partial_node=remote_info.partial_node)
+
+        if not wait_for_results:
+            return None
+        results, stdout, stderr = xpr.get_results()
+
+        sys.stdout.write(stdout)
+        sys.stderr.write(stderr)
+
+        # no outputs to rename since everything should be in run_dir
+        xpr.mark_processed()
+
+        return results
+
+    if need_to_fit:
+        if not run_dir.exists():
+            run_dir.mkdir(parents=True)
+
+        use_params = dict(params)
+
+        if dry_run:
+            use_params['dry_run'] = None
+
+        # the code below needs to know an unfortunate amount about the inner workings of ace_fit.jl
+        # here we rely on knowledge of the --outfile_base
+        use_params['outfile_base'] = ace_file_base
+        use_params['outfile_format'] = formats
+
+        use_params['key'] = [ [k, ref_property_prefix + v] for k,v in [('E', 'energy'), ('F', 'forces'), ('V', 'virial')] ]
+        use_params['_repeat_key'] = True
+
+        # configs need to be in memory so they can be modified with stress -> virial
+        fitting_configs = list(fitting_configs)
```

Comment:
why not `ConfigSet_in.in_memory()`?

Reply:
Maybe not much use after all - I initially thought having a configsetin would slightly simplify reading/writing ~10 lines below

Reply:
I think it's just copied from code that predates that function.  Let me think if there's any reason to prefer one or the other syntax.

Reply:
Below we pass that variable to `ase.io.write()`, and I thought it's safest to do it as a list. Comment explains this in 683738a.

I guess the real question is whether `to_memory()` is every worth it, but that's a matter for a different PR.